src/helpers.js

```diff
@@ -901,9 +901,10 @@ export function vmSupportsExternalSnapshots(config, vm, storagePools) {
 
     // Currently external snapshots works only for disks where we can specify a local path of a newly created disk snapshot file
     // This rules out all disks which are not file-based, or pool-based where pool is of type "dir"
+    // or media-less disks
     const supportedPools = storagePools.filter(pool => pool.type === "dir" && pool.connectionName === vm.connectionName);
     if (!disks.every(disk =>
-        disk.type === "file" ||
+        (disk.type === "file" && disk.source?.file) ||
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1440-20240212-085014-bd28d833-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)